src/logql-query.ts

```diff
@@ -0,0 +1,48 @@
+import { notEmptyString, trim } from './value-utils';
+
+export class LogQLQuery {
+  selectors: Array<string> = [];
+  pipeline: Array<string> = [];
+
+  constructor(query: string) {
+    const parts = query.match(/(.*){([^}]*)}(.*)/);
```

Comment:
You could remove the first `(.*)` since you're not actually using it. Alternatively, you could force it to match the whole string by adding `^` at the start and `$` at the end (`/^.*{([^}]*)}(.*)$/`).